apisix/plugins/ai-proxy/schema.lua

```diff
@@ -0,0 +1,167 @@
+            enum = {"header", "param"}
+        },
+        name = {
+            type = "string",
+            description = "Name of the param/header carrying Authorization or API key.",
+            minLength = 1,
+        },
+        value = {
+            type = "string",
+            description = "Full auth-header/param value.",
+            minLength = 1,
+             -- TODO encrypted = true,
+        },
+    },
+    required = { "source", "name", "value" },
+    additionalProperties = false,
+}
+
+local model_options_schema = {
+    description = "Key/value settings for the model",
+    type = "object",
+    properties = {
+        max_tokens = {
+            type = "integer",
+            description = "Defines the max_tokens, if using chat or completion models.",
+            default = 256
+
+        },
+        input_cost = {
+            type = "number",
+            description = "Defines the cost per 1M tokens in your prompt.",
+            minimum = 0
+
+        },
+        output_cost = {
+            type = "number",
+            description = "Defines the cost per 1M tokens in the output of the AI.",
+            minimum = 0
+
+        },
+        temperature = {
+            type = "number",
+            description = "Defines the matching temperature, if using chat or completion models.",
+            minimum = 0.0,
+            maximum = 5.0,
+
+        },
+        top_p = {
+            type = "number",
+            description = "Defines the top-p probability mass, if supported.",
+            minimum = 0,
+            maximum = 1,
+
+        },
+        upstream_host = {
+            type = "string",
+            description = "To be specified to override the host of the AI provider",
+        },
+        upstream_port = {
+            type = "integer",
+            description = "To be specified to override the AI provider port",
+
+        },
+        upstream_path = {
+            type = "string",
+            description = "To be specified to override the URL to the AI provider endpoints",
+        },
+        response_streaming = {
+            description = "Stream response by SSE",
+            type = "boolean",
+            default = false,
+        }
+    }
+}
+
+local model_schema = {
+    type = "object",
+    properties = {
+        provider = {
+            type = "string",
+            description = "AI provider request format - kapisix translates "
+                .. "requests to and from the specified backend compatible formats.",
+            oneOf = { "openai" }, -- add more providers later
+
+        },
+        name = {
+            type = "string",
+            description = "Model name to execute.",
+        },
+        options = model_options_schema,
+    },
+    required = {"provider", "name"}
+}
+
+_M.plugin_schema = {
+    type = "object",
+    properties = {
+        route_type = {
+            type = "string",
+            enum = { "llm/chat", "llm/completions", "passthrough" }
```

Comment:
I don't see the need to support the old and inflexible `llm/completions` API.

Reply:
Yea I will remove it since the chat API can do what the completion api does.